sherpa/models/model.py

```diff
@@ -46,12 +46,12 @@ def cache_model(cls, pars, xlo, *args, **kwargs):
         digest = ''
         if use_caching:
 
-            data = [ numpy.array(pars).tostring(), str(kwargs.get('integrate',0)), 
+            data = [ numpy.array(pars).tostring(), kwargs.get('integrate', b'0'),
```

Comment:
````   
bmap = {True: b'1', False: b'0'}
data = [numpy.array(pars).tostring(), bmap.get(kwargs.get('integrate', False), b'0'),
                    numpy.asarray(xlo).tostring()]
````

This should work in all cases: it always returns `b'0' or b'1'`, including the case in which `integrate` is not a boolean.